tests/flow/test_hnsw_tiered.py

```diff
@@ -0,0 +1,563 @@
+        return hnsw_index
+    
+    def generate_queries(self, num_queries):
+        queries = self.rng.random((num_queries, self.dim)) 
+        return np.float32(queries) if self.data_type == VecSimType_FLOAT32 else queries
+    
+    def get_vectors_memory_size(self):
+        data_type_size = 4 if self.data_type == VecSimType_FLOAT32 else 8
+        return bytes_to_mega(self.num_vectors * self.dim * data_type_size)
+           
+def create_tiered_index(is_multi: bool, num_per_label = 1):
+    indices_ctx = IndexCtx(data_size=50000, is_multi=is_multi, num_per_label=num_per_label)
+    num_elements = indices_ctx.num_labels
+    
+    threads_num = TieredIndex.get_threads_num()
+    
+    # Initialize time measurements to apply assert at the end.
+    tiered_index_time = 0
+    hnsw_index_time = 0
+    
+    # Create the tiered index
+    index = indices_ctx.tiered_index
+    
+    _, bf_dur, end_add_time = indices_ctx.populate_index(index)
+    
+    index.wait_for_index()
+    tiered_index_time = bf_dur + time.time() - end_add_time
+    
+    assert index.hnsw_label_count() == num_elements
+    
+    # Measure insertion to tiered index.
+    print(f"Insert {num_elements} insertion jobs took {round_ms(bf_dur)} ms")
+    print(f"Done insertion jobs executions using {threads_num} threads took {round_ms(tiered_index_time)} ms")
+    
+    # Measure total memory of the tiered index.
+    tiered_memory = bytes_to_mega(index.index_memory())
+    
+    print(f"total memory of tiered index = {tiered_memory} MB")
+    
+    hnsw_index = HNSWIndex(indices_ctx.hnsw_params)
+    _, hnsw_index_time, _ = indices_ctx.populate_index(hnsw_index)
+
+    print(f"Insert {num_elements} vectors to sync hnsw index took {round_(hnsw_index_time)} s")   
+    hnsw_memory = bytes_to_mega(hnsw_index.index_memory())
+    print(f"total memory of hnsw index = {hnsw_memory} MB")
+    
+    # The index memory should be at least as the total memory of the vectors.
+    assert hnsw_memory > indices_ctx.get_vectors_memory_size()
+    
+    # Tiered index memory should be greater than HNSW index memory.
+    assert tiered_memory > hnsw_memory
+    execution_time_ratio = hnsw_index_time / tiered_index_time
+    print(f"with {threads_num} threads, insertion runtime is {round_(execution_time_ratio)} times better \n")
+    
+
+def search_insert(is_multi: bool, num_per_label = 1):
+    indices_ctx = IndexCtx(data_size=100000, is_multi=is_multi, num_per_label=num_per_label)
+    index = indices_ctx.tiered_index
+
+    num_labels = indices_ctx.num_labels
+    
+    query_data = indices_ctx.generate_queries(num_queries=1)
+    
+    # Add vectors to the flat index.
+    bf_index = indices_ctx.init_and_populate_flat_index()
+    
+    # Start background insertion to the tiered index.
+    index_start, _, _ = indices_ctx.populate_index(index)
+    
+    correct = 0
+    k = 10
+    searches_number = 0
+    print(f"hnsw size = {index.hnsw_label_count()}")
```

Comment:
For multi this is not accurate, since for some labels we may have more than one vector in HNSW index at that point...

Reply:
This is why we're getting the labels count and not indexSize
The printing is indeed misleading